engines/experiment/manager/adapter_test.go

```diff
@@ -0,0 +1,257 @@
+package manager_test
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/gojek/turing/engines/experiment/manager"
+	"github.com/gojek/turing/engines/experiment/manager/mocks"
+)
+
+func TestStandardExperimentMethods(t *testing.T) {
```

Comment:
Is there any particular reason why do you merge test cases for all the methods in a single test function? Can we have each function tested separately for easier debugging when something fails?

Reply:
I chose to do this because they all have the same set up function (for `GetEngineInfo`).

Separated them out now.

Later, we could integrate: https://github.com/stretchr/testify, which may also help address your other comment: https://github.com/gojek/turing/pull/118#discussion_r748848904

But I will keep this for a future MR.